libcst/codemod/commands/convert_namedtuple_to_dataclass.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# pyre-strict
+from typing import List, Optional, Sequence
+
+import libcst as cst
+from libcst.codemod import VisitorBasedCodemodCommand
+from libcst.codemod.visitors import AddImportsVisitor, RemoveImportsVisitor
+from libcst.metadata import (
+    ProviderT,
+    QualifiedName,
+    QualifiedNameProvider,
+    QualifiedNameSource,
+)
+
+
+class ConvertNamedTupleToDataclassCommand(VisitorBasedCodemodCommand):
+    """
+    Convert NamedTuple class declarations to Python 3.7 dataclasses.
+
+    This only performs a conversion at the class declaration level.
+    It does not perform type annotation conversions, nor does it convert
+    NamedTuple-specific attributes and methods.
+    """
+
+    DESCRIPTION: str = "Convert legacy NamedTuple class declarations to Python 3.7 dataclasses."
+    METADATA_DEPENDENCIES: Sequence[ProviderT] = (QualifiedNameProvider,)
+
+    MODULE: str = "typing"
+    OBJECT: str = "NamedTuple"
+
+    def leave_ClassDef(
+        self, original_node: cst.ClassDef, updated_node: cst.ClassDef
+    ) -> cst.ClassDef:
+        new_bases: List[cst.Arg] = []
+        namedtuple_base: Optional[cst.Arg] = None
+        qualified_namedtuple: QualifiedName = QualifiedName(
+            name=f"{self.MODULE}.{self.OBJECT}", source=QualifiedNameSource.IMPORT
+        )
```

Comment:
This is immutable and creating it has no side effects; any reason not to just make the entire `QualifiedName` object a class attribute, and not have to construct it anew each time we leave a class definition?

Reply:
Very good point